src/tests/ftest/telemetry/engine_events.py

```diff
@@ -0,0 +1,270 @@
+
+    def check_rank_stopped(self, restart_rank, dmg_command):
+        """Check the restart_rank is stopped.
+
+        If it hasn't been stopped, fail the test. This method is created to avoid the pylint error.
+
+        Args:
+            restart_rank (int): Restarted rank.
+            dmg_command (DmgCommand): DmgCommand object.
+        """
+        rank_stopped = False
+        for count in range(3):
+            time.sleep(5)
+            query_out = dmg_command.system_query()
+            for member in query_out["response"]["members"]:
+                if member["rank"] == restart_rank:
+                    if member["state"] == "stopped" or member["state"] == "excluded":
+                        rank_stopped = True
+                        break
+            if rank_stopped:
+                self.log.info("Rank %d is stopped. count = %d", restart_rank, count)
+                break
+            self.log.info("Rank %d is not stopped. Check again. count = %d", restart_rank, count)
+        if not rank_stopped:
+            self.fail(f"Rank {restart_rank} didn't stop!")
+
+    def check_rank_restarted(self, restart_rank, dmg_command):
+        """Check the restart_rank is restarted.
+
+        If it hasn't been restarted, fail the test. This method is created to avoid the pylint
+        error.
+
+        Args:
+            restart_rank (int): Restarted rank.
+            dmg_command (DmgCommand): DmgCommand object.
+        """
+        rank_restarted = False
+        for count in range(3):
+            time.sleep(5)
+            query_out = dmg_command.system_query()
+            for member in query_out["response"]["members"]:
+                if member["rank"] == restart_rank:
+                    if member["state"] == "joined":
+                        rank_restarted = True
+                        break
+            if rank_restarted:
+                self.log.info("Rank %d is joined. count = %d", restart_rank, count)
+                break
+            self.log.info("Rank %d is not joined. Check again. count = %d", restart_rank, count)
+        if not rank_restarted:
+            self.fail(f"Rank {restart_rank} didn't restart!")
+
+    def test_engine_events(self):
+        """Test engine-related events after an engine is restarted.
+
+        1. Check that all ranks have started.
+        2. Gather initial data for the required metrics for all the server nodes.
+        3. Stop a rank.
+        4. Verify the desired rank has stopped successfully.
+        5. Restart the stopped rank after several seconds.
+        6. Verify the desired rank restarted successfully.
+        7. Gather final data for required metrics with the same command as in step 2.
+        8. Compare the before and after values.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=telemetry
+        :avocado: tags=test_engine_events
```

Comment:
Please add the class name as a tag. This will be very useful later for some work to automatically determine test tags based on files modified.
```suggestion
        :avocado: tags=EngineEvents,test_engine_events
```

Reply:
I forgot about that. Done.